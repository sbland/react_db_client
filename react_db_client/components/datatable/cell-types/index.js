export { DataTableCellButton } from './DataTableCellButton';
export { DataTableCellEntity } from './DataTableCellEntity';
export { DataTableCellLink } from './DataTableCellLink';
export { DataTableCellNumber } from './DataTableCellNumber';
export { DataTableCellPopup } from './DataTableCellPopup';
export { DataTableCellReadOnly } from './DataTableCellReadOnly';
export { DataTableCellSelect } from './DataTableCellSelect';
export { DataTableCellText } from './DataTableCellText';
export { DataTableCellToggle } from './DataTableCellToggle';

export { defaultComponentMap } from './default-component-map';
