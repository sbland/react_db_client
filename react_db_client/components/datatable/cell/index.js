export { Cell } from './cell';
export { DataTableCellHoverWrap } from './cell-wrappers';
