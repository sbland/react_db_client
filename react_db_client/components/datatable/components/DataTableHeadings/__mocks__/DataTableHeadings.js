/* eslint-disable react/prop-types */
/* eslint-disable arrow-body-style */
/* eslint-disable no-unused-vars */
import React from 'react';

export const DataTableHeadingMenu = ({
  handleAddFilter,
  setSortBy,
  headingData,
  handleHideColumn,
}) => {
  return <div className="dataTableHeadingMenu">Data Table Heading Menu</div>;
};

export const DataTableHeading = ({ title }) => (
  <div className="dataTableHeading">
    <div className="heading">{title}</div>
  </div>
);

export const DataTableHeadings = ({
  headingsDataList,
  setSortBy,
  handleHideColumn,
  columnWidths,
  handleAddFilter,
  showFilterBtns,
  tableWidth,
}) => {
  return <div className="dataTableHeadings">mock data table headings</div>;
};

export default DataTableHeadings;
