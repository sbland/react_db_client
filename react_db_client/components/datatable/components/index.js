// export { DataTableCellHoverWrap } from './DataTableCell';
export { DataTableTotals } from './DataTableTotals';
export { DataTableHeadings } from './DataTableHeadings';
export { defaultComponentMap } from './default-component-map';
