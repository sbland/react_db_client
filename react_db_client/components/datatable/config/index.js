export {
  dataTableDefaultConfig,
  DataTableContext,
  DataTableConfigConnector,
} from './DataTableConfig';
