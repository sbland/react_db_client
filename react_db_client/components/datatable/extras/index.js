export { RowErrors } from './errorTypes';
export { headingDataShape, tableDataShape } from './inputDataShapes';
export * from './demoData';