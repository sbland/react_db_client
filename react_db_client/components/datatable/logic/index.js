// export { useDataManager } from './data-manager/DataManager';
// export { useConditionalStylingManager } from './ConditionalStylingManager/ConditionalStylingManager';
// export { useSelectionManager } from './SelectionManagerHook';
export { RowStyleContext } from './RowStyleContext/RowStyleContext';
