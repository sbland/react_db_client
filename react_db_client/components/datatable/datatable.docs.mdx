---
description: 'A datatable spreadsheet component.'
labels: ['data', 'ui']
---

import { Datatable } from './datatable';

## React Component for displaying and editing table data

A simple spreadsheet component

