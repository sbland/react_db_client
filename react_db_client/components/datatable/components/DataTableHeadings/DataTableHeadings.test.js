import React from 'react';
import { mount, shallow } from 'enzyme';
import { DataTableHeadings } from './DataTableHeadings';

import { demoHeadingsData } from '@react_db_client/components.datatable.extras';
import { DataTableContext, dataTableDefaultConfig } from '@react_db_client/components.datatable.logic';

describe('DataTableHeading', () => {
  it('Renders', () => {
    mount(
      <DataTableHeadings
        headingsDataList={demoHeadingsData}
        setsortBy={() => {}}
        handleHideColumn={() => {}}
        handleAddFilter={() => {}}
        columnWidths={[
          100, // append column width for edit btns
          ...demoHeadingsData.map(() => 110),
        ]}
      />
    );
  });
  it('Matches Snapshot', () => {
    const component = shallow(
      <DataTableContext.Provider value={dataTableDefaultConfig}>
        <DataTableHeadings
          headingsDataList={demoHeadingsData}
          setsortBy={() => {}}
          handleHideColumn={() => {}}
          handleAddFilter={() => {}}
          columnWidths={[
            100, // append column width for edit btns
            ...demoHeadingsData.map(() => 110),
          ]}
        />
      </DataTableContext.Provider>
    )
      .find(DataTableHeadings)
      .dive();
    const tree = component.debug();
    expect(tree).toMatchSnapshot();
  });
});
