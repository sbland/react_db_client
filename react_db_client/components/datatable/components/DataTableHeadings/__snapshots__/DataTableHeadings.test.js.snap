// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeading Matches Snapshot 1`] = `
"<div className=\\"dataTable_headingsOutsideWrap\\">
  <div className=\\"dataTable_headingsInsideWrap\\" style={{...}}>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={100} style={{...}} handleHover={[Function: handleHover]} disabled={false} />
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"UID\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Name\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"description\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Count\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Def\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Multiplier\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Read Only\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Hidden\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Hidden Number\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Toggle\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"ToggleR\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Expression\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Is Validated\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Invalid\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"InvalidB\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Select\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Internal Room\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Long Heading which is really really long\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Custom\\" />
    </DataTableCellHoverWrap>
    <DataTableCellHoverWrap className=\\"dataTableHeading_cellWrap\\" columnWidth={110} handleHover={[Function: handleHover]} style={{...}} disabled={false}>
      <DataTableHeading title=\\"Custom Eval\\" />
    </DataTableCellHoverWrap>
  </div>
</div>"
`;
